src/librustdoc/test.rs

```diff
@@ -736,12 +762,22 @@ impl Tester for Collector {
         let linker = self.linker.clone();
         let edition = config.edition.unwrap_or(self.edition);
         let persist_doctests = self.persist_doctests.clone();
+        let runtool = self.runtool.clone();
+        let runtool_args = self.runtool_args.clone();
+        let target = self.target.clone();
+        let target_str = target.to_string();
 
         debug!("Creating test {}: {}", name, test);
         self.tests.push(testing::TestDescAndFn {
             desc: testing::TestDesc {
                 name: testing::DynTestName(name.clone()),
-                ignore: config.ignore,
+                ignore: match config.ignore {
+                    Ignore::All => true,
+                    Ignore::None => false,
+                    Ignore::Some(ref ignores) => {
+                        ignores.iter().any(|s| target_str.contains(s))
```

Comment:
Is there a better method of testing the `ignore-` string than this? I was hoping we could reuse [compiletest's filters](https://rust-lang.github.io/rustc-guide/tests/adding.html#ignoring-tests), which don't seem like they could work with a substring search.

Reply:
I have not investigated leveraging any existing machinery from compiltest's filters, although in the worst case it would not be too much trouble to extend the flags to handle only and some other target details.

Reply:
I don't think the compiletest filters are particularly well-suited to stabilizing. They just define a custom list of keywords for matching platforms. Since compiletest is for internal use and this is not, I think we can put a little more thought into this.

I do agree that substring matching on the target name is a little strange. It would be nice to somehow hook into Rust's existing conditional compilation story, which is `cfg`.

Reply:
In the end I think this is sufficient for an initial implementation of an unstable feature, and others can expand this as the need arises.